x/tier/keeper/abci.go

```diff
@@ -0,0 +1,94 @@
+package keeper
+
+import (
+	"context"
+
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	appparams "github.com/sourcenetwork/sourcehub/app/params"
+	"github.com/sourcenetwork/sourcehub/x/tier/types"
+)
+
+// BeginBlocker claims tier module staking rewards every N blocks.
+// 2% of the claimed rewards is sent to the developer pool (DeveloperPoolFee).
+// 1% is sent to the insurance pool (InsurancePoolFee) if insurance pool balance is below InsurancePoolThreshold,
+// otherwise the InsurancePoolFee (1%) is also sent to the developer pool.
+// Remaining 97% of the rewards is burned.
+func (k *Keeper) BeginBlocker(ctx context.Context) error {
+	params := k.GetParams(ctx)
+
+	// Process rewards every N blocks
+	height := sdk.UnwrapSDKContext(ctx).BlockHeight()
+	if height%params.ProcessRewardsInterval != 0 {
+		return nil
+	}
```

Comment:
praise: Good catch to add this. Hopefully we will be able to measure the perf impact this function has and be able to adjust the interval to accomodate if necessary.